scripts/core/program.yml

```diff
@@ -68,7 +19,7 @@ members:
     - type: uint32_t
       name: count
       desc: "[in] the number of entries in pMetadatas, if count is greater than zero then pMetadatas must not be null."
-    - type: const $x_program_metadata_t*
+    - type: const $x_device_binary_property_t*
```

Comment:
This is not my favorite change. Seems much less clear reading this here, also its not just a binary property its also passed to `CreateWithIL`.

Reply:
Yeah I think this is poorly names. I was copying the existing style that is in PI - is `piProgramCreateWithBinary` equivalent to `urProgramCreateWithIL`?

Reply:
> Yeah I think this is poorly names. I was copying the existing style that is in PI - is `piProgramCreateWithBinary` equivalent to `urProgramCreateWithIL`?

No, `piProgramCreateWithBinary` == `urProgramCreateWithBinary` and `piProgramCreate` == `urProgramCreateWithIL` although when I ported the metadata structs I added them to both of the UR entry points. Currently the CUDA plugin passes PTX, which is an IL, to `piProgramCreateWithBinary` but logically it should be `urProgramCreateWithIL` which is why I added it to both entrypoints.

Reply:
My mistake thanks. Do you think it can just be called something generic like `ur_metadata_property_t`?